### DISCUSSION TITLE
Leveraging this extension for documenting screens?

### DISCUSSION THREAD

**Discussion by Lopson** [renpy/vscode-language-renpy#457] [Q&A]
Hey everyone! I'm trying to figure out what the best way would be to document screens in my own Ren'Py project, and for the life of me I can't quite figure out how to do it seeing as the usual Python docstrings are not parsable when inside a screen's definition.

In the process of figuring this out, I noticed that the built-in screens that are made available by default, like the `say` one, do show documentation when you hover your mouse cursor over the screen's name! Which got me thinking...

Is there a way for me to leverage this functionality of this extension to also neatly document my own custom screens in my project? I know this is a long shot, but wanted to ask nonetheless.

Thanks for your time! 😊

**Comment by duckdoom4** [comment]
Hi, that sounds very interesting. Could you show me an example of what kind of documentation you'd like to expose? Is it provided in your source files?

If so I think it should be possible to extract that information with this extension (with some extra work required on the extension side to extract and expose the information).

Edit:
Reading your post again, I think I understand what you're looking for. The documentation provided by the extension is sourced from the renpy SDK. There is a file called `renpy.json` that contains all the functions and their doc strings. 

Technically you could edit this file, but since it's auto generated and part of the extension it's pretty much impossible to add custom ones in there, but I'd be very interested in trying to figure out a way to get this implemented. Preferably directly in source (like python doc strings)

**Reply by Lopson** [reply]
Hi hi! Yeah, so my thinking was something along these lines... Let's say we have a screen like so:

`screen inventory_screen(target_jump = True, handle_quick_menu = True, show_close_button = True, return_actions = [Return()]):`

I want to document it like so:

> The inventory screen where the player can go over the items they've acquired or present evidence. This screen is either called from a script or invoked by the Inventory button usually shown in the ADV interface during normal gameplay.
> 
> :param target_jump: Tells the "Present Evidence" button to jump to the script label attached to an inventory item if True or to do the actions listed in `return_actions` if False.
> 
> :param handle_quick_menu: Tells the screen whether or not it should hide the quick menu. Does this if set to True; does nothing if False.
> 
> :param show_close_button: Controls whether to show the close button or not. Used for inventory challenges where we don't want the player to be able to exit this screen without presenting an item. Shows the button if True; hides it if set to False.
> 
> :param return_actions: List of screen actions to take **only** when `target_jump` is set to False.

My idea was:
1. Either allow us to somehow write docstrings directly below the declaration of the screen but still within its scope, just like regular Python docstrings, OR
2. Define a standardized folder in which we'd be able to put RST files that could document all kinds of things, like screens.

Option no.2 seems more doable with the very cursory knowledge I have of this project. 🤔
